`zmq.h` and `linux/tipc.h`: No suchfile or directory in docker build

Docker build for enterprise_gateway raised following errors.
```
Sending build context to Docker daemon  90.62kB
Step 1/12 : FROM elyra/spark-py:v2.4.0
 ---> 2938dec60816
Step 2/12 : RUN apk add --no-cache build-base libffi-dev openssl-dev python-dev curl krb5-dev
 ---> Using cache
 ---> be8542621a5a
Step 3/12 : COPY jupyter_enterprise_gateway*.whl /tmp
 ---> e5311596b86f
Step 4/12 : RUN pip install cffi send2trash /tmp/jupyter_enterprise_gateway*.whl &&     pip install requests --upgrade &&       rm -f /tmp/jupyter_enterprise_gateway*.whl
 ---> Running in d30b41dfc4f2
Collecting cffi
  Downloading https://files.pythonhosted.org/packages/bc/81/47bd0404f2cb5363edb371e3b15da6387b5e9b80122e5b81be8b8f411e9b/cffi-1.12.1.tar.gz (453kB)
Collecting send2trash
  Downloading https://files.pythonhosted.org/packages/13/2e/ea40de0304bb1dc4eb309de90aeec39871b9b7c4bd30f1a3cdcb3496f5c0/Send2Trash-1.5.0.tar.gz
Processing /tmp/jupyter_enterprise_gateway-2.0.0.dev1-py2.py3-none-any.whl
Collecting pycparser (from cffi)
  Downloading https://files.pythonhosted.org/packages/68/9e/49196946aee219aead1290e00d1e7fdeab8567783e83e1b9ab5585e6206a/pycparser-2.19.tar.gz (158kB)
Collecting docker>=3.5.0 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/7e/3c/b610f22b170b0f8fe4d8f78974878e116562389f666f99e6549567eb9d87/docker-3.7.0-py2.py3-none-any.whl (133kB)
Collecting yarn-api-client>=0.3.0 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/67/6a/a15801cb7eb496040988b1f58f5a6b1ac622cd7c82eb7d72170753851dab/yarn_api_client-0.3.2-py2.py3-none-any.whl
Collecting jupyter-client>=5.2.0 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/3b/c3/3043fe9ffd140d03c9d091a056794ccdc427c56ec19b8eea74f9ea0a498f/jupyter_client-5.2.4-py2.py3-none-any.whl (89kB)
Collecting jupyter-kernel-gateway>=2.1.0 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/9f/2a/a202ea047acbe4b716cb73e9f8a12f4595b503d41c2ca909d9e33e686c9d/jupyter_kernel_gateway-2.1.0-py2.py3-none-any.whl (46kB)
Collecting pexpect>=4.2.0 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/89/e6/b5a1de8b0cc4e07ca1b305a4fcc3f9806025c1b651ea302646341222f88b/pexpect-4.6.0-py2.py3-none-any.whl (57kB)
Collecting traitlets>=4.2.0 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/93/d6/abcb22de61d78e2fc3959c964628a5771e47e7cc60d53e9342e21ed6cc9a/traitlets-4.3.2-py2.py3-none-any.whl (74kB)
Collecting pyzmq>=17.0.0 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/64/8d/78975da77627fd863c08e8ea3c7cebce7e51bed2936be5118de6b0050638/pyzmq-18.0.0.tar.gz (1.2MB)
Collecting notebook<6.0,>=5.3.0 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/f8/a6/dbdf0954d073ab2cf5421cad20334d1e4c780da37205c04522c045298f67/notebook-5.7.4-py2.py3-none-any.whl (9.0MB)
Collecting pycrypto>=2.6.1 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/60/db/645aa9af249f059cc3a368b118de33889219e0362141e75d4eaf6f80f163/pycrypto-2.6.1.tar.gz (446kB)
Collecting tornado>=4.2.0 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/e6/78/6e7b5af12c12bdf38ca9bfe863fcaf53dc10430a312d0324e76c1e5ca426/tornado-5.1.1.tar.gz (516kB)
Collecting paramiko>=2.1.2 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/cf/ae/94e70d49044ccc234bfdba20114fa947d7ba6eb68a2e452d89b920e62227/paramiko-2.4.2-py2.py3-none-any.whl (193kB)
Collecting jupyter-core>=4.4.0 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/1d/44/065d2d7bae7bebc06f1dd70d23c36da8c50c0f08b4236716743d706762a8/jupyter_core-4.4.0-py2.py3-none-any.whl (126kB)
Collecting requests<3.0,>=2.7 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/7d/e3/20f3d364d6c8e5d2353c72a67778eb189176f08e873c9900e10c0287b84b/requests-2.21.0-py2.py3-none-any.whl (57kB)
Collecting kubernetes>=4.0.0 (from jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/40/eb/4d6a80db84ac24c867c94fbf16e6f26db9780f5232f46ddd2e5539b42205/kubernetes-8.0.1-py2.py3-none-any.whl (1.3MB)
Collecting backports.ssl-match-hostname>=3.5; python_version < "3.5" (from docker>=3.5.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/ff/2b/8265224812912bc5b7a607c44bf7b027554e1b9775e9ee0de8032e3de4b2/backports.ssl_match_hostname-3.7.0.1.tar.gz
Collecting docker-pycreds>=0.4.0 (from docker>=3.5.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/f5/e8/f6bd1eee09314e7e6dee49cbe2c5e22314ccdb38db16c9fc72d2fa80d054/docker_pycreds-0.4.0-py2.py3-none-any.whl
Collecting ipaddress>=1.0.16; python_version < "3.3" (from docker>=3.5.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/fc/d0/7fc3a811e011d4b388be48a0e381db8d990042df54aa4ef4599a31d39853/ipaddress-1.0.22-py2.py3-none-any.whl
Collecting websocket-client>=0.32.0 (from docker>=3.5.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/26/2d/f749a5c82f6192d77ed061a38e02001afcba55fe8477336d26a950ab17ce/websocket_client-0.54.0-py2.py3-none-any.whl (200kB)
Collecting six>=1.4.0 (from docker>=3.5.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting python-dateutil>=2.1 (from jupyter-client>=5.2.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/41/17/c62faccbfbd163c7f57f3844689e3a78bae1f403648a6afb1d0866d87fbb/python_dateutil-2.8.0-py2.py3-none-any.whl (226kB)
Collecting ptyprocess>=0.5 (from pexpect>=4.2.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/d1/29/605c2cc68a9992d18dada28206eeada56ea4bd07a239669da41674648b6f/ptyprocess-0.6.0-py2.py3-none-any.whl
Collecting ipython-genutils (from traitlets>=4.2.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/fa/bc/9bd3b5c2b4774d5f33b2d544f1460be9df7df2fe42f352135381c347c69a/ipython_genutils-0.2.0-py2.py3-none-any.whl
Collecting decorator (from traitlets>=4.2.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/f1/cd/7c8240007e9716b14679bc217a1baefa4432aa30394f7e2ec40a52b1a708/decorator-4.3.2-py2.py3-none-any.whl
Collecting enum34; python_version == "2.7" (from traitlets>=4.2.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
Collecting jinja2 (from notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/7f/ff/ae64bacdfc95f27a016a7bed8e8686763ba4d277a78ca76f32659220a731/Jinja2-2.10-py2.py3-none-any.whl (126kB)
Collecting nbformat (from notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/da/27/9a654d2b6cc1eaa517d1c5a4405166c7f6d72f04f6e7eea41855fe808a46/nbformat-4.4.0-py2.py3-none-any.whl (155kB)
Collecting nbconvert (from notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/b8/39/1e67fea74dc9577cc49f9863fe3ec824e525d1304ab6027d95a94cd586f5/nbconvert-5.4.1-py2.py3-none-any.whl (407kB)
Collecting terminado>=0.8.1 (from notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/2e/20/a26211a24425923d46e1213b376a6ee60dc30bcdf1b0c345e2c3769deb1c/terminado-0.8.1-py2.py3-none-any.whl
Collecting ipykernel (from notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/00/47/764e4fa1b1b89598426b8d79b1c4fbe8042432621b0f8e1991aeb3c24806/ipykernel-4.10.0-py2-none-any.whl (110kB)
Collecting prometheus-client (from notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/4c/bd/b42db3ec90ffc6be805aad09c1cea4bb13a620d0cd4b21aaa44d13541d71/prometheus_client-0.6.0.tar.gz
Collecting futures (from tornado>=4.2.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/2d/99/b2c4e9d5a30f6471e410a146232b4118e697fa3ffc06d6a65efde84debd0/futures-3.2.0-py2-none-any.whl
Collecting singledispatch (from tornado>=4.2.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/c5/10/369f50bcd4621b263927b0a1519987a04383d4a98fb10438042ad410cf88/singledispatch-3.4.0.3-py2.py3-none-any.whl
Collecting backports_abc>=0.4 (from tornado>=4.2.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/7d/56/6f3ac1b816d0cd8994e83d0c4e55bc64567532f7dc543378bd87f81cebc7/backports_abc-0.5-py2.py3-none-any.whl
Collecting pyasn1>=0.1.7 (from paramiko>=2.1.2->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/7b/7c/c9386b82a25115cccf1903441bba3cbadcfae7b678a20167347fa8ded34c/pyasn1-0.4.5-py2.py3-none-any.whl (73kB)
Collecting bcrypt>=3.1.3 (from paramiko>=2.1.2->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/ce/3a/3d540b9f5ee8d92ce757eebacf167b9deedb8e30aedec69a2a072b2399bb/bcrypt-3.1.6.tar.gz (42kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
Collecting cryptography>=1.5 (from paramiko>=2.1.2->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/69/ed/5e97b7f54237a9e4e6291b6e52173372b7fa45ca730d36ea90b790c0059a/cryptography-2.5.tar.gz (487kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
Collecting pynacl>=1.0.1 (from paramiko>=2.1.2->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/61/ab/2ac6dea8489fa713e2b4c6c5b549cc962dd4a842b5998d9e80cf8440b7cd/PyNaCl-1.3.0.tar.gz (3.4MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
Collecting chardet<3.1.0,>=3.0.2 (from requests<3.0,>=2.7->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl (133kB)
Collecting idna<2.9,>=2.5 (from requests<3.0,>=2.7->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl (58kB)
Collecting certifi>=2017.4.17 (from requests<3.0,>=2.7->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/9f/e0/accfc1b56b57e9750eba272e24c4dddeac86852c2bebd1236674d7887e8a/certifi-2018.11.29-py2.py3-none-any.whl (154kB)
Collecting urllib3<1.25,>=1.21.1 (from requests<3.0,>=2.7->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/62/00/ee1d7de624db8ba7090d1226aebefab96a2c71cd5cfa7629d6ad3f61b79e/urllib3-1.24.1-py2.py3-none-any.whl (118kB)
Collecting requests-oauthlib (from kubernetes>=4.0.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/c2/e2/9fd03d55ffb70fe51f587f20bcf407a6927eb121de86928b34d162f0b1ac/requests_oauthlib-1.2.0-py2.py3-none-any.whl
Collecting pyyaml>=3.12 (from kubernetes>=4.0.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/9e/a3/1d13970c3f36777c583f136c136f804d70f500168edc1edea6daa7200769/PyYAML-3.13.tar.gz (270kB)
Collecting google-auth>=1.0.1 (from kubernetes>=4.0.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/c5/9b/ed0516cc1f7609fb0217e3057ff4f0f9f3e3ce79a369c6af4a6c5ca25664/google_auth-1.6.3-py2.py3-none-any.whl (73kB)
Collecting adal>=1.0.2 (from kubernetes>=4.0.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/00/72/53dce9e4f5d6c1aa57b8d408cb34dff1969ecbf10ab7e678f32c5e0e2397/adal-1.2.1-py2.py3-none-any.whl (52kB)
Requirement already satisfied: setuptools>=21.0.0 in /usr/lib/python2.7/site-packages (from kubernetes>=4.0.0->jupyter-enterprise-gateway==2.0.0.dev1) (40.6.2)
Collecting MarkupSafe>=0.23 (from jinja2->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/ac/7e/1b4c2e05809a4414ebce0892fe1e32c14ace86ca7d50c70f00979ca9b3a3/MarkupSafe-1.1.0.tar.gz
Collecting jsonschema!=2.5.0,>=2.4 (from nbformat->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/77/de/47e35a97b2b05c2fadbec67d44cfcdcd09b8086951b331d82de90d2912da/jsonschema-2.6.0-py2.py3-none-any.whl
Collecting pandocfilters>=1.4.1 (from nbconvert->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/4c/ea/236e2584af67bb6df960832731a6e5325fd4441de001767da328c33368ce/pandocfilters-1.4.2.tar.gz
Collecting mistune>=0.8.1 (from nbconvert->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/09/ec/4b43dae793655b7d8a25f76119624350b4d65eb663459eb9603d7f1f0345/mistune-0.8.4-py2.py3-none-any.whl
Collecting entrypoints>=0.2.2 (from nbconvert->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/ac/c6/44694103f8c221443ee6b0041f69e2740d89a25641e62fb4f2ee568f2f9c/entrypoints-0.3-py2.py3-none-any.whl
Collecting testpath (from nbconvert->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/be/a4/162f9ebb6489421fe46dcca2ae420369edfee4b563c668d93cb4605d12ba/testpath-0.4.2-py2.py3-none-any.whl (163kB)
Collecting defusedxml (from nbconvert->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/87/1c/17f3e3935a913dfe2a5ca85fa5ccbef366bfd82eb318b1f75dadbf0affca/defusedxml-0.5.0-py2.py3-none-any.whl
Collecting pygments (from nbconvert->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/13/e5/6d710c9cf96c31ac82657bcfb441df328b22df8564d58d0c4cd62612674c/Pygments-2.3.1-py2.py3-none-any.whl (849kB)
Collecting bleach (from nbconvert->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/ab/05/27e1466475e816d3001efb6e0a85a819be17411420494a1e602c36f8299d/bleach-3.1.0-py2.py3-none-any.whl (157kB)
Collecting ipython>=4.0.0 (from ipykernel->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/b0/88/d996ab8be22cea1eaa18baee3678a11265e18cf09974728d683c51102148/ipython-5.8.0-py2-none-any.whl (760kB)
Collecting asn1crypto>=0.21.0 (from cryptography>=1.5->paramiko>=2.1.2->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/ea/cd/35485615f45f30a510576f1a56d1e0a7ad7bd8ab5ed7cdc600ef7cd06222/asn1crypto-0.24.0-py2.py3-none-any.whl (101kB)
Collecting oauthlib>=3.0.0 (from requests-oauthlib->kubernetes>=4.0.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/16/95/699466b05b72b94a41f662dc9edf87fda4289e3602ecd42d27fcaddf7b56/oauthlib-3.0.1-py2.py3-none-any.whl (142kB)
Collecting pyasn1-modules>=0.2.1 (from google-auth>=1.0.1->kubernetes>=4.0.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/da/98/8ddd9fa4d84065926832bcf2255a2b69f1d03330aa4d1c49cc7317ac888e/pyasn1_modules-0.2.4-py2.py3-none-any.whl (66kB)
Collecting cachetools>=2.0.0 (from google-auth>=1.0.1->kubernetes>=4.0.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/39/2b/d87fc2369242bd743883232c463f28205902b8579cb68dcf5b11eee1652f/cachetools-3.1.0-py2.py3-none-any.whl
Collecting rsa>=3.1.4 (from google-auth>=1.0.1->kubernetes>=4.0.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/02/e5/38518af393f7c214357079ce67a317307936896e961e35450b70fad2a9cf/rsa-4.0-py2.py3-none-any.whl
Collecting PyJWT>=1.0.0 (from adal>=1.0.2->kubernetes>=4.0.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/87/8b/6a9f14b5f781697e51259d81657e6048fd31a113229cf346880bb7545565/PyJWT-1.7.1-py2.py3-none-any.whl
Collecting functools32; python_version == "2.7" (from jsonschema!=2.5.0,>=2.4->nbformat->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/c5/60/6ac26ad05857c601308d8fb9e87fa36d0ebf889423f47c3502ef034365db/functools32-3.2.3-2.tar.gz
Collecting configparser>=3.5; python_version == "2.7" (from entrypoints>=0.2.2->nbconvert->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/e9/b1/98e5e3094aac1c0efaeb208fb45f9b42eb1d5681b4c5dac77b05d0fec71a/configparser-3.7.1-py2-none-any.whl
Collecting webencodings (from bleach->nbconvert->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/f4/24/2a3e3df732393fed8b3ebf2ec078f05546de641fe1b667ee316ec1dcf3b7/webencodings-0.5.1-py2.py3-none-any.whl
Collecting prompt-toolkit<2.0.0,>=1.0.4 (from ipython>=4.0.0->ipykernel->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/d1/b0/1a6c262da35c779dd79550137aa7c298a424987240a28792ec5ccf48f848/prompt_toolkit-1.0.15-py2-none-any.whl (247kB)
Collecting backports.shutil-get-terminal-size; python_version == "2.7" (from ipython>=4.0.0->ipykernel->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/7d/cd/1750d6c35fe86d35f8562091737907f234b78fdffab42b29c72b1dd861f4/backports.shutil_get_terminal_size-1.0.0-py2.py3-none-any.whl
Collecting pathlib2; python_version == "2.7" or python_version == "3.3" (from ipython>=4.0.0->ipykernel->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/2a/46/c696dcf1c7aad917b39b875acdc5451975e3a9b4890dca8329983201c97a/pathlib2-2.3.3-py2.py3-none-any.whl
Collecting simplegeneric>0.8 (from ipython>=4.0.0->ipykernel->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/3d/57/4d9c9e3ae9a255cd4e1106bb57e24056d3d0709fc01b2e3e345898e49d5b/simplegeneric-0.8.1.zip
Collecting pickleshare (from ipython>=4.0.0->ipykernel->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/9a/41/220f49aaea88bc6fa6cba8d05ecf24676326156c23b991e80b3f2fc24c77/pickleshare-0.7.5-py2.py3-none-any.whl
Collecting wcwidth (from prompt-toolkit<2.0.0,>=1.0.4->ipython>=4.0.0->ipykernel->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/7e/9f/526a6947247599b084ee5232e4f9190a38f398d7300d866af3ab571a5bfe/wcwidth-0.1.7-py2.py3-none-any.whl
Collecting scandir; python_version < "3.5" (from pathlib2; python_version == "2.7" or python_version == "3.3"->ipython>=4.0.0->ipykernel->notebook<6.0,>=5.3.0->jupyter-enterprise-gateway==2.0.0.dev1)
  Downloading https://files.pythonhosted.org/packages/16/2a/557af1181e6b4e30254d5a6163b18f5053791ca66e251e77ab08887e8fe3/scandir-1.9.0.tar.gz
Installing collected packages: pycparser, cffi, send2trash, backports.ssl-match-hostname, six, docker-pycreds, ipaddress, chardet, idna, certifi, urllib3, requests, websocket-client, docker, yarn-api-client, ipython-genutils, decorator, enum34, traitlets, jupyter-core, futures, singledispatch, backports-abc, tornado, python-dateutil, pyzmq, jupyter-client, MarkupSafe, jinja2, functools32, jsonschema, nbformat, pandocfilters, mistune, configparser, entrypoints, testpath, defusedxml, pygments, webencodings, bleach, nbconvert, ptyprocess, terminado, wcwidth, prompt-toolkit, backports.shutil-get-terminal-size, pexpect, scandir, pathlib2, simplegeneric, pickleshare, ipython, ipykernel, prometheus-client, notebook, jupyter-kernel-gateway, pycrypto, pyasn1, bcrypt, asn1crypto, cryptography, pynacl, paramiko, oauthlib, requests-oauthlib, pyyaml, pyasn1-modules, cachetools, rsa, google-auth, PyJWT, adal, kubernetes, jupyter-enterprise-gateway
  Running setup.py install for pycparser: started
    Running setup.py install for pycparser: finished with status 'done'
  Running setup.py install for cffi: started
    Running setup.py install for cffi: finished with status 'done'
  Running setup.py install for send2trash: started
    Running setup.py install for send2trash: finished with status 'done'
  Running setup.py install for backports.ssl-match-hostname: started
    Running setup.py install for backports.ssl-match-hostname: finished with status 'done'
  Running setup.py install for tornado: started
    Running setup.py install for tornado: finished with status 'done'
  Running setup.py install for pyzmq: started
    Running setup.py install for pyzmq: finished with status 'error'
    Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-GzlP_J/pyzmq/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-oSOkYm/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-2.7
    creating build/lib.linux-x86_64-2.7/zmq
    copying zmq/error.py -> build/lib.linux-x86_64-2.7/zmq
    copying zmq/decorators.py -> build/lib.linux-x86_64-2.7/zmq
    copying zmq/_future.py -> build/lib.linux-x86_64-2.7/zmq
    copying zmq/__init__.py -> build/lib.linux-x86_64-2.7/zmq
    creating build/lib.linux-x86_64-2.7/zmq/devices
    copying zmq/devices/proxydevice.py -> build/lib.linux-x86_64-2.7/zmq/devices
    copying zmq/devices/proxysteerabledevice.py -> build/lib.linux-x86_64-2.7/zmq/devices
    copying zmq/devices/monitoredqueuedevice.py -> build/lib.linux-x86_64-2.7/zmq/devices
    copying zmq/devices/basedevice.py -> build/lib.linux-x86_64-2.7/zmq/devices
    copying zmq/devices/__init__.py -> build/lib.linux-x86_64-2.7/zmq/devices
    copying zmq/devices/monitoredqueue.py -> build/lib.linux-x86_64-2.7/zmq/devices
    creating build/lib.linux-x86_64-2.7/zmq/eventloop
    copying zmq/eventloop/future.py -> build/lib.linux-x86_64-2.7/zmq/eventloop
    copying zmq/eventloop/_deprecated.py -> build/lib.linux-x86_64-2.7/zmq/eventloop
    copying zmq/eventloop/ioloop.py -> build/lib.linux-x86_64-2.7/zmq/eventloop
    copying zmq/eventloop/__init__.py -> build/lib.linux-x86_64-2.7/zmq/eventloop
    copying zmq/eventloop/zmqstream.py -> build/lib.linux-x86_64-2.7/zmq/eventloop
    creating build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado
    copying zmq/eventloop/minitornado/util.py -> build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado
    copying zmq/eventloop/minitornado/concurrent.py -> build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado
    copying zmq/eventloop/minitornado/log.py -> build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado
    copying zmq/eventloop/minitornado/stack_context.py -> build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado
    copying zmq/eventloop/minitornado/ioloop.py -> build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado
    copying zmq/eventloop/minitornado/__init__.py -> build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado
    creating build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado/platform
    copying zmq/eventloop/minitornado/platform/common.py -> build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado/platform
    copying zmq/eventloop/minitornado/platform/posix.py -> build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado/platform
    copying zmq/eventloop/minitornado/platform/auto.py -> build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado/platform
    copying zmq/eventloop/minitornado/platform/interface.py -> build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado/platform
    copying zmq/eventloop/minitornado/platform/__init__.py -> build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado/platform
    copying zmq/eventloop/minitornado/platform/windows.py -> build/lib.linux-x86_64-2.7/zmq/eventloop/minitornado/platform
    creating build/lib.linux-x86_64-2.7/zmq/ssh
    copying zmq/ssh/tunnel.py -> build/lib.linux-x86_64-2.7/zmq/ssh
    copying zmq/ssh/forward.py -> build/lib.linux-x86_64-2.7/zmq/ssh
    copying zmq/ssh/__init__.py -> build/lib.linux-x86_64-2.7/zmq/ssh
    creating build/lib.linux-x86_64-2.7/zmq/green
    copying zmq/green/core.py -> build/lib.linux-x86_64-2.7/zmq/green
    copying zmq/green/device.py -> build/lib.linux-x86_64-2.7/zmq/green
    copying zmq/green/poll.py -> build/lib.linux-x86_64-2.7/zmq/green
    copying zmq/green/__init__.py -> build/lib.linux-x86_64-2.7/zmq/green
    creating build/lib.linux-x86_64-2.7/zmq/green/eventloop
    copying zmq/green/eventloop/ioloop.py -> build/lib.linux-x86_64-2.7/zmq/green/eventloop
    copying zmq/green/eventloop/__init__.py -> build/lib.linux-x86_64-2.7/zmq/green/eventloop
    copying zmq/green/eventloop/zmqstream.py -> build/lib.linux-x86_64-2.7/zmq/green/eventloop
    creating build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_pubsub.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_z85.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_ioloop.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_message.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_zmqstream.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_error.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_decorators.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_imports.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_monqueue.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_reqrep.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_monitor.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_socket.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_context.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_proxy_steerable.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_retry_eintr.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_auth.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_etc.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_future.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_poll.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_version.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_ssh.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_includes.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_security.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_device.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_log.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_cffi_backend.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_win32_shim.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_multipart.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/__init__.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_constants.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_pair.py -> build/lib.linux-x86_64-2.7/zmq/tests
    copying zmq/tests/test_draft.py -> build/lib.linux-x86_64-2.7/zmq/tests
    creating build/lib.linux-x86_64-2.7/zmq/sugar
    copying zmq/sugar/constants.py -> build/lib.linux-x86_64-2.7/zmq/sugar
    copying zmq/sugar/attrsettr.py -> build/lib.linux-x86_64-2.7/zmq/sugar
    copying zmq/sugar/socket.py -> build/lib.linux-x86_64-2.7/zmq/sugar
    copying zmq/sugar/version.py -> build/lib.linux-x86_64-2.7/zmq/sugar
    copying zmq/sugar/tracker.py -> build/lib.linux-x86_64-2.7/zmq/sugar
    copying zmq/sugar/poll.py -> build/lib.linux-x86_64-2.7/zmq/sugar
    copying zmq/sugar/frame.py -> build/lib.linux-x86_64-2.7/zmq/sugar
    copying zmq/sugar/context.py -> build/lib.linux-x86_64-2.7/zmq/sugar
    copying zmq/sugar/__init__.py -> build/lib.linux-x86_64-2.7/zmq/sugar
    copying zmq/sugar/stopwatch.py -> build/lib.linux-x86_64-2.7/zmq/sugar
    creating build/lib.linux-x86_64-2.7/zmq/backend
    copying zmq/backend/select.py -> build/lib.linux-x86_64-2.7/zmq/backend
    copying zmq/backend/__init__.py -> build/lib.linux-x86_64-2.7/zmq/backend
    creating build/lib.linux-x86_64-2.7/zmq/backend/cffi
    copying zmq/backend/cffi/error.py -> build/lib.linux-x86_64-2.7/zmq/backend/cffi
    copying zmq/backend/cffi/utils.py -> build/lib.linux-x86_64-2.7/zmq/backend/cffi
    copying zmq/backend/cffi/_poll.py -> build/lib.linux-x86_64-2.7/zmq/backend/cffi
    copying zmq/backend/cffi/devices.py -> build/lib.linux-x86_64-2.7/zmq/backend/cffi
    copying zmq/backend/cffi/_cffi.py -> build/lib.linux-x86_64-2.7/zmq/backend/cffi
    copying zmq/backend/cffi/constants.py -> build/lib.linux-x86_64-2.7/zmq/backend/cffi
    copying zmq/backend/cffi/message.py -> build/lib.linux-x86_64-2.7/zmq/backend/cffi
    copying zmq/backend/cffi/socket.py -> build/lib.linux-x86_64-2.7/zmq/backend/cffi
    copying zmq/backend/cffi/context.py -> build/lib.linux-x86_64-2.7/zmq/backend/cffi
    copying zmq/backend/cffi/__init__.py -> build/lib.linux-x86_64-2.7/zmq/backend/cffi
    creating build/lib.linux-x86_64-2.7/zmq/backend/cython
    copying zmq/backend/cython/__init__.py -> build/lib.linux-x86_64-2.7/zmq/backend/cython
    creating build/lib.linux-x86_64-2.7/zmq/auth
    copying zmq/auth/thread.py -> build/lib.linux-x86_64-2.7/zmq/auth
    copying zmq/auth/certs.py -> build/lib.linux-x86_64-2.7/zmq/auth
    copying zmq/auth/base.py -> build/lib.linux-x86_64-2.7/zmq/auth
    copying zmq/auth/ioloop.py -> build/lib.linux-x86_64-2.7/zmq/auth
    copying zmq/auth/__init__.py -> build/lib.linux-x86_64-2.7/zmq/auth
    creating build/lib.linux-x86_64-2.7/zmq/log
    copying zmq/log/handlers.py -> build/lib.linux-x86_64-2.7/zmq/log
    copying zmq/log/__init__.py -> build/lib.linux-x86_64-2.7/zmq/log
    creating build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/constant_names.py -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/garbage.py -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/strtypes.py -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/z85.py -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/monitor.py -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/sixcerpt.py -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/jsonapi.py -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/interop.py -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/__init__.py -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/win32.py -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/devices/monitoredqueue.pxd -> build/lib.linux-x86_64-2.7/zmq/devices
    copying zmq/backend/cffi/_cdefs.h -> build/lib.linux-x86_64-2.7/zmq/backend/cffi
    copying zmq/backend/cffi/_verify.c -> build/lib.linux-x86_64-2.7/zmq/backend/cffi
    copying zmq/backend/cython/context.pxd -> build/lib.linux-x86_64-2.7/zmq/backend/cython
    copying zmq/backend/cython/libzmq.pxd -> build/lib.linux-x86_64-2.7/zmq/backend/cython
    copying zmq/backend/cython/checkrc.pxd -> build/lib.linux-x86_64-2.7/zmq/backend/cython
    copying zmq/backend/cython/message.pxd -> build/lib.linux-x86_64-2.7/zmq/backend/cython
    copying zmq/backend/cython/socket.pxd -> build/lib.linux-x86_64-2.7/zmq/backend/cython
    copying zmq/backend/cython/constants.pxi -> build/lib.linux-x86_64-2.7/zmq/backend/cython
    copying zmq/backend/cython/constant_enums.pxi -> build/lib.linux-x86_64-2.7/zmq/backend/cython
    copying zmq/utils/buffers.pxd -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/zmq_constants.h -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/getpid_compat.h -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/zmq_compat.h -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/ipcmaxlen.h -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/pyversion_compat.h -> build/lib.linux-x86_64-2.7/zmq/utils
    copying zmq/utils/mutex.h -> build/lib.linux-x86_64-2.7/zmq/utils
    running build_ext
    running configure
    Did not find libzmq via pkg-config.
    {'libraries': ['zmq'], 'extra_link_args': [], 'runtime_library_dirs': [], 'library_dirs': [], 'include_dirs': []}
    gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -c build/temp.linux-x86_64-2.7/scratch/check_sys_un.c -o build/temp.linux-x86_64-2.7/scratch/check_sys_un.o
    gcc build/temp.linux-x86_64-2.7/scratch/check_sys_un.o -o build/temp.linux-x86_64-2.7/scratch/check_sys_un
    ************************************************
    Configure: Autodetecting ZMQ settings...
        Custom ZMQ dir:
    creating build/temp.linux-x86_64-2.7/scratch/tmp
    gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -c /tmp/timer_createF_vacI.c -o build/temp.linux-x86_64-2.7/scratch/tmp/timer_createF_vacI.o
    /tmp/timer_createF_vacI.c:1:1: warning: return type defaults to 'int' [-Wimplicit-int]
     main (int argc, char **argv) {
     ^~~~
    /tmp/timer_createF_vacI.c: In function 'main':
    /tmp/timer_createF_vacI.c:2:5: warning: implicit declaration of function 'timer_create' [-Wimplicit-function-declaration]
         timer_create();
         ^~~~~~~~~~~~
    gcc build/temp.linux-x86_64-2.7/scratch/tmp/timer_createF_vacI.o -o build/temp.linux-x86_64-2.7/scratch/a.out
    gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -Izmq/utils -Izmq/backend/cython -Izmq/devices -c build/temp.linux-x86_64-2.7/scratch/vers.c -o build/temp.linux-x86_64-2.7/scratch/vers.o
    build/temp.linux-x86_64-2.7/scratch/vers.c:4:17: fatal error: zmq.h: No such file or directory
     #include "zmq.h"
                     ^
    compilation terminated.

    error: command 'gcc' failed with exit status 1

    Failed with default libzmq, trying again with /usr/local
    {'libraries': ['zmq'], 'extra_link_args': [], 'runtime_library_dirs': ['/usr/local/lib'], 'library_dirs': ['/usr/local/lib'], 'include_dirs': ['/usr/local/include']}
    ************************************************
    Configure: Autodetecting ZMQ settings...
        Custom ZMQ dir:       /usr/local
    gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -c /tmp/timer_createnzBgNU.c -o build/temp.linux-x86_64-2.7/scratch/tmp/timer_createnzBgNU.o
    /tmp/timer_createnzBgNU.c:1:1: warning: return type defaults to 'int' [-Wimplicit-int]
     main (int argc, char **argv) {
     ^~~~
    /tmp/timer_createnzBgNU.c: In function 'main':
    /tmp/timer_createnzBgNU.c:2:5: warning: implicit declaration of function 'timer_create' [-Wimplicit-function-declaration]
         timer_create();
         ^~~~~~~~~~~~
    Assembler messages:
    Fatal error: can't create build/temp.linux-x86_64-2.7/scratch/tmp/timer_createnzBgNU.o: No such file or directory
    gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -I/usr/local/include -Izmq/utils -Izmq/backend/cython -Izmq/devices -c build/temp.linux-x86_64-2.7/scratch/vers.c -o build/temp.linux-x86_64-2.7/scratch/vers.o
    build/temp.linux-x86_64-2.7/scratch/vers.c:4:17: fatal error: zmq.h: No such file or directory
     #include "zmq.h"
                     ^
    compilation terminated.

    error: command 'gcc' failed with exit status 1

    ************************************************
    Warning: Couldn't find an acceptable libzmq on the system.

    If you expected pyzmq to link against an installed libzmq, please check to make sure:

        * You have a C compiler installed
        * A development version of Python is installed (including headers)
        * A development version of ZMQ >= 3.2 is installed (including headers)
        * If ZMQ is not in a default location, supply the argument --zmq=<path>
        * If you did recently install ZMQ to a default location,
          try rebuilding the ld cache with `sudo ldconfig`
          or specify zmq's location with `--zmq=/usr/local`

    You can skip all this detection/waiting nonsense if you know
    you want pyzmq to bundle libzmq as an extension by passing:

        `--zmq=bundled`

    I will now try to build libzmq as a Python extension
    unless you interrupt me (^C) in the next 10 seconds...

     1...
    ************************************************
    Using bundled libzmq
    already have bundled/zeromq
    attempting ./configure to generate platform.hpp
    Warning: failed to configure libzmq:
    /bin/sh: ./configure: No such file or directory

    staging platform.hpp from: buildutils/include_linux
    ************************************************
    checking for timer_create
    creating build/temp.linux-x86_64-2.7/tmp
    gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -c /tmp/timer_createmdQBot.c -o build/temp.linux-x86_64-2.7/tmp/timer_createmdQBot.o
    /tmp/timer_createmdQBot.c:1:1: warning: return type defaults to 'int' [-Wimplicit-int]
     main (int argc, char **argv) {
     ^~~~
    /tmp/timer_createmdQBot.c: In function 'main':
    /tmp/timer_createmdQBot.c:2:5: warning: implicit declaration of function 'timer_create' [-Wimplicit-function-declaration]
         timer_create();
         ^~~~~~~~~~~~
    gcc build/temp.linux-x86_64-2.7/tmp/timer_createmdQBot.o -o build/temp.linux-x86_64-2.7/a.out
    ok
    ************************************************
    building 'zmq.libzmq' extension
    creating build/temp.linux-x86_64-2.7/buildutils
    creating build/temp.linux-x86_64-2.7/bundled
    creating build/temp.linux-x86_64-2.7/bundled/zeromq
    creating build/temp.linux-x86_64-2.7/bundled/zeromq/src
    gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -DZMQ_HAVE_CURVE=1 -DZMQ_USE_TWEETNACL=1 -DZMQ_USE_EPOLL=1 -DZMQ_IOTHREADS_USE_EPOLL=1 -DZMQ_POLL_BASED_ON_POLL=1 -Ibundled/zeromq/include -Ibundled -I/usr/include/python2.7 -c buildutils/initlibzmq.c -o build/temp.linux-x86_64-2.7/buildutils/initlibzmq.o
    gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -DZMQ_HAVE_CURVE=1 -DZMQ_USE_TWEETNACL=1 -DZMQ_USE_EPOLL=1 -DZMQ_IOTHREADS_USE_EPOLL=1 -DZMQ_POLL_BASED_ON_POLL=1 -Ibundled/zeromq/include -Ibundled -I/usr/include/python2.7 -c bundled/zeromq/src/udp_engine.cpp -o build/temp.linux-x86_64-2.7/bundled/zeromq/src/udp_engine.o
    gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -DZMQ_HAVE_CURVE=1 -DZMQ_USE_TWEETNACL=1 -DZMQ_USE_EPOLL=1 -DZMQ_IOTHREADS_USE_EPOLL=1 -DZMQ_POLL_BASED_ON_POLL=1 -Ibundled/zeromq/include -Ibundled -I/usr/include/python2.7 -c bundled/zeromq/src/tcp.cpp -o build/temp.linux-x86_64-2.7/bundled/zeromq/src/tcp.o
    gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -DZMQ_HAVE_CURVE=1 -DZMQ_USE_TWEETNACL=1 -DZMQ_USE_EPOLL=1 -DZMQ_IOTHREADS_USE_EPOLL=1 -DZMQ_POLL_BASED_ON_POLL=1 -Ibundled/zeromq/include -Ibundled -I/usr/include/python2.7 -c bundled/zeromq/src/signaler.cpp -o build/temp.linux-x86_64-2.7/bundled/zeromq/src/signaler.o
    gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -DZMQ_HAVE_CURVE=1 -DZMQ_USE_TWEETNACL=1 -DZMQ_USE_EPOLL=1 -DZMQ_IOTHREADS_USE_EPOLL=1 -DZMQ_POLL_BASED_ON_POLL=1 -Ibundled/zeromq/include -Ibundled -I/usr/include/python2.7 -c bundled/zeromq/src/tipc_connecter.cpp -o build/temp.linux-x86_64-2.7/bundled/zeromq/src/tipc_connecter.o
    In file included from bundled/zeromq/src/tipc_connecter.cpp:46:0:
    bundled/zeromq/src/tipc_address.hpp:43:24: fatal error: linux/tipc.h: No such file or directory
     #include <linux/tipc.h>
                            ^
    compilation terminated.
    error: command 'gcc' failed with exit status 1

```

Solution: Install packages; zeromq-dev for zmq.h and linux-headers for linux/tipc.h
```
RUN apk add --no-cache build-base libffi-dev openssl-dev python-dev curl krb5-dev linux-headers zeromq-dev
```

--- a/etc/docker/enterprise-gateway/Dockerfile
+++ b/etc/docker/enterprise-gateway/Dockerfile
@@ -1,6 +1,6 @@
 FROM elyra/spark-py:v2.4.0
 
-RUN apk add --no-cache build-base libffi-dev openssl-dev python-dev curl krb5-dev
+RUN apk add --no-cache build-base libffi-dev openssl-dev python-dev curl krb5-dev linux-headers zeromq-dev
 
 # Install Enterprise Gateway wheel and kernelspecs
 COPY jupyter_enterprise_gateway*.whl /tmp

